src/core/Song.cpp

```diff
@@ -1546,5 +1548,20 @@ void Song::setKeymap(unsigned int index, std::shared_ptr<Keymap> newMap)
 	Engine::audioEngine()->doneChangeInModel();
 }
 
+void Song::processMetronome(size_t bufferOffset)
+{
+	const auto currentPlayMode = playMode();
+	const auto supported = currentPlayMode == Song::PlayMode::MidiClip
+		|| currentPlayMode == Song::PlayMode::Song
+		|| currentPlayMode == Song::PlayMode::Pattern;
+
+	if (!supported || isExporting()) { return; } 
+	m_metronome.processTick(currentTick(), ticksPerBar(), m_timeSigModel.getNumerator(), bufferOffset);
```

Comment:
I'm wondering if it would be better to remove this function and call `Metronome::processTick` directly. Should the metronome be allowed to play even when exporting? Or even in all cases?

Reply:
Seems like a good idea. More decoupling

Reply:
Yeah this function seems like a wrapper for an if else looking at the diff.

Reply:
I'll leave it for now since I'm uncertain.